services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/attachment/ApplicationAttachmentMigrationSchedulerTest.kt

```diff
@@ -0,0 +1,209 @@
+package fi.hel.haitaton.hanke.attachment
+
+import fi.hel.haitaton.hanke.attachment.application.ApplicationAttachmentContentService.Companion.generateBlobPath
+import fi.hel.haitaton.hanke.attachment.application.ApplicationAttachmentMigrationScheduler
+import fi.hel.haitaton.hanke.attachment.application.ApplicationAttachmentMigrationScheduler.Companion.LOCK_NAME
+import fi.hel.haitaton.hanke.attachment.application.ApplicationAttachmentMigrator
+import fi.hel.haitaton.hanke.attachment.application.ApplicationAttachmentWithContent
+import fi.hel.haitaton.hanke.attachment.common.AttachmentContentNotFoundException
+import fi.hel.haitaton.hanke.configuration.LockService
+import fi.hel.haitaton.hanke.factory.ApplicationAttachmentFactory
+import io.mockk.Called
+import io.mockk.checkUnnecessaryStub
+import io.mockk.clearAllMocks
+import io.mockk.confirmVerified
+import io.mockk.every
+import io.mockk.justRun
+import io.mockk.mockk
+import io.mockk.verify
+import io.mockk.verifyOrder
+import java.util.UUID
+import java.util.concurrent.locks.Lock
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.integration.jdbc.lock.JdbcLockRegistry
+
+class ApplicationAttachmentMigrationSchedulerTest {
+    private val migrator: ApplicationAttachmentMigrator = mockk()
+    private val jdbcLockRegistry: JdbcLockRegistry = mockk()
+    private val lockService: LockService = LockService(jdbcLockRegistry)
+
+    private val scheduler =
+        ApplicationAttachmentMigrationScheduler(
+            lockService,
+            migrator,
+        )
+
+    @BeforeEach
+    fun clearMocks() {
+        clearAllMocks()
+    }
+
+    @AfterEach
+    fun confirmMocks() {
+        checkUnnecessaryStub()
+        confirmVerified(
+            jdbcLockRegistry,
+            migrator,
+        )
+    }
+
+    @Nested
+    inner class ScheduleMigrate {
+        @Test
+        fun `does nothing if all content has been migrated`() {
+            mockLocking(true)
+            every { migrator.findAttachmentWithDatabaseContent() } returns null
+
+            scheduler.scheduleMigrate()
+
+            verifyOrder {
+                jdbcLockRegistry.obtain(LOCK_NAME)
+                migrator.findAttachmentWithDatabaseContent()
+            }
+            verify(exactly = 0) { migrator.migrate(any()) }
+            verify(exactly = 0) { migrator.setBlobPathAndCleanup(any(), any()) }
```

Comment:
VerifySequence checks that no other calls were made to the mocks mentioned in the block.
```suggestion
            verifySequence {
                jdbcLockRegistry.obtain(LOCK_NAME)
                migrator.findAttachmentWithDatabaseContent()
            }
```
This also checks the order of the calls, even though it's not documented directly in the function's KDoc.

Reply:
Thank you for teaching me a new thing again! Fixed.